src/lib/watch-processor.ts

```diff
@@ -106,12 +106,22 @@ async function runBinding(binding: Binding, capabilityNamespaces: string[], igno
             if (!obj.metadata?.deletionTimestamp) {
               return;
             }
+
+            let shouldRemoveFinalizer: boolean | void | undefined = true;
             try {
-              await binding.finalizeCallback?.(obj);
+              shouldRemoveFinalizer = await binding.finalizeCallback?.(obj);
 
-              // irrespective of callback success / failure, remove pepr finalizer
+              // if not opt'ed out of / if in error state, remove pepr finalizer
             } finally {
-              await removeFinalizer(binding, obj);
+              const peprFinal = "pepr.dev/finalizer";
+              const meta = obj.metadata!;
+              const resource = `${meta.namespace || "ClusterScoped"}/${meta.name}`;
+
+              // [ true, void, undefined ] SHOULD remove finalizer
+              // [ false ] should NOT remove finalizer
+              shouldRemoveFinalizer === false
+                ? Log.debug({ obj }, `Skip removal of finalizer '${peprFinal}' from '${resource}'`)
```

Comment:
```suggestion
                ? Log.debug({ obj }, `Skipping removal of finalizer '${peprFinal}' from '${resource}'`)
```

Reply:
Don't most of our log lines use a -ing verb? If so, we may want to make this "Skipping" instead of "Skip"